docs/contributing/feature-flags.md

```diff
@@ -20,6 +20,65 @@ highlights:
 - Environments (production, QA, and development for now) exist to segment flag states further. This
   will be automatic based on where code is running.
 
+## When to use flags
+
+Feature flags are a powerful tool for controlling the rollout and lifecycle of features, and there
+are several scenarios when you should consider using them:
+
+### Progressive rollouts
+
+Use feature flags when you need to gradually release a feature to users. This allows you to:
+
+- Start with a small percentage of users to validate stability
+- Monitor performance and user feedback before full deployment
+- Quickly disable the feature if issues arise without requiring a code rollback
+
+### A / B testing and experimentation
+
+Feature flags enable controlled experiments by:
+
+- Serving different variations of a feature to different user segments
+- Collecting metrics to determine which variation performs better
+- Making data-driven decisions about feature implementation
```

Comment:
To my knowledge we don't actually do this today and it would probably not be a dev decision. No harm in mentioning it as a potential use case but I suggest moving it to the bottom of this list to de-emphasise it.